apollo-backend/src/main/java/io/logz/apollo/controllers/BlockerDefinitionController.java

```diff
@@ -168,9 +168,13 @@ public void updateBlockerDefinition(int id, String name, Integer environmentId,
                                         Boolean isActive, String blockerTypeName, String blockerJsonConfiguration, Req req) {
         String userEmail = req.token().get("_user").toString();
         logger.info("User: {} has just updated blocker, id:{}, name:{}", userEmail, id, name);
-
         BlockerDefinition blockerDefinition = blockerDefinitionDao.getBlockerDefinition(id);
 
+        if (!blockerDefinition.getBlockerTypeName().equals(blockerTypeName)) {
+            assignJsonResponseToReq(req, HttpStatus.BAD_REQUEST, String.format("You can not edit the blocker type. attempt to covert {} blocker into {} blocker", blockerDefinition.getBlockerTypeName(), blockerTypeName));
```

Comment:
why? I think it might be by design? (possibility to change blocker type)
please ask @mayhuberman

Reply:
@matvey-mtn Different blocker types have different Json fields...

Reply:
@matvey-mtn So it is error prone.

Reply:
We should enable editing the blocker type as well

Reply:
@matvey-mtn @mayhuberman. May, unfortunately

Reply:
@maorshap let's schedule a call and discuss it